webapp/src/blocks.tsx

```diff
@@ -1287,11 +1288,43 @@ export class Editor extends toolboxeditor.ToolboxEditor {
         this.showFlyoutInternal_(this.flyoutXmlList);
     }
 
-    private showFlyoutInternal_(xmlList: Element[]) {
+    private flyouts: pxt.Map<Blockly.VerticalFlyout> = {};
```

Comment:
When do we clear this cache?

Reply:
So far, never. I can't think of a case where we'd want to. This cache doesn't grow unbounded, there's only one one entry per toolbox category. We refresh the contents of the flyouts in these caches when appropriate. Some flyouts that change a lot like variables we don't even cache.

Reply:
It's not exactly a cache. We're just maintaining multiple flyouts at once, and they're mutable.